core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -69,9 +70,7 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
         old
       } else {
         val newDir = new File(localDirs(dirId), "%02x".format(subDirId))
-        if (!newDir.exists() && !newDir.mkdir()) {
-          throw new IOException(s"Failed to create local dir in $newDir.")
-        }
+        Files.createDirectory(newDir.toPath)
```

Comment:
This code looks like a regression. Previously, if the directory exists, we don't throw `IOException`.
Now, this PR will throw `FileAlreadyExistsException` if the directory exists.

Reply:
Yeah, either `exists()` -> `Files.createDirectory` or just `Files.createDirectories` would do the trick. The side-effect of latter is that it would also create non-existence parent directories.

Reply:
+1 for @HeartSaVioR 's advice.